libass/ass_render.c

```diff
@@ -1826,38 +1760,24 @@ static int is_new_bm_run(GlyphInfo *info, GlyphInfo *last)
         last->blur != info->blur ||
         last->shadow_x != info->shadow_x ||
         last->shadow_y != info->shadow_y ||
-        last->frx != info->frx ||
-        last->fry != info->fry ||
-        last->frz != info->frz ||
-        last->fax != info->fax ||
-        last->fay != info->fay ||
-        last->scale_x != info->scale_x ||
-        last->scale_y != info->scale_y ||
         last->border_style != info->border_style ||
-        last->border_x != info->border_x ||
-        last->border_y != info->border_y ||
-        last->hspacing != info->hspacing ||
-        last->italic != info->italic ||
-        last->bold != info->bold ||
-        last->flags != info->flags){
-        return 1;
-    }
-    return 0;
+        (last->border_x || last->border_y) != (info->border_x || info->border_y) ||
+        !last->border != !info->border;
```

Comment:
What’s up with all of this? `is_new_bm_run` must stay as it is for VSFilter compatibility. I think there may be one or two values where we still technically differ from VSFilter, but certainly the bulk of it matches exactly.